### PR TITLE
Adding code arena readme file.

### PR DESCRIPTION
**Pull request by mijovic**

### CODE REVIEW

File: other/CodeArenaReadme.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+| Glossary| |
+|-------------------------------|------------------------------------------------------|
+| Backing token| Any token that can be deposited to earn some interest via landing or staking. Currently we support ETH, DAI, and USDC |
+| YBT Yield bearing token| Landing or staking token, currently we support stETH, aTokens, and cTokens |
+| TempusPool| Holds all locked YieldBearingToken, and is used to mint Principals and Yields on deposits, and to burn them on redemption in exchange for YieldBearingToken |
+| Principals| Zero coupon bond redeemable for 1 Backing token after maturity |
+| Yields| Zero coupon bond redeemable after maturity for amount of Yield 1 Principal accrued for the lifetime of the pool |
+| Tempus AMM| Tempus AMM implementation based on Stable Pools from Balancer v2. It is used for trading Principals against Yields |
+| LP token| Token that represent share in Liquidity Pool |
+| Tempus Controller| Entry point to protocol. Used for all user facing actions related to Tempus Pool |
```
Comment (mijovic) [inline other/CodeArenaReadme.md:21]: This should show up as table. I added some of the mostly used terms here

File: other/CodeArenaReadme.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+This repo will be made public before the start of the contest. (C4 delete this line when made public)
+
+| Glossary| |
+|-------------------------------|------------------------------------------------------|
+| Backing token| Any token that can be deposited to earn some interest via landing or staking. Currently we support ETH, DAI, and USDC |
+| YBT Yield bearing token| Landing or staking token, currently we support stETH, aTokens, and cTokens |
+| TempusPool| Holds all locked YieldBearingToken, and is used to mint Principals and Yields on deposits, and to burn them on redemption in exchange for YieldBearingToken |
+| Principals| Zero coupon bond redeemable for 1 Backing token after maturity |
+| Yields| Zero coupon bond redeemable after maturity for amount of Yield 1 Principal accrued for the lifetime of the pool |
```
Comment (RedFox20) [inline other/CodeArenaReadme.md:18]:
```suggestion
| Yields| Zero-coupon bond redeemable to BackingTokens after maturity for amount of yield accrued during the lifetime of the pool. Example: 10% yield would exchange 0.1 BackingToken for each Yield share |
```

File: other/CodeArenaReadme.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Currently, there is no easy way to obtain a fixed yield or otherwise speculate on the returns in a capital-efficient way. This is where Tempus steps in, allowing users to perform two different use cases, each one offering a unique value proposition:
+- Buy and sell interest rate protection using any supported Yield Bearing Token (such as stETH, cDai and others).
+- Earn swap fees as a liquidity provider by depositing any supported Yield Bearing Token (thus earning additional yield on top of yield earned through other yield farming protocols).
+
+Users can deposit backing or yield bearing token to mint equal amounts of Principals and Yields. More information can be found in [deposit docs](https://docs.tempus.finance/docs/tempuspool/deposit).
+
+Redemption has two different flows:
+- Early redemption (this is possible only with equal amount of Principals and Yields)
+- Mature redemption (user can deposit any combination of Principals and Yields after maturity)
+Redemption can be to backing or yield bearing token. You can find more in [redemption docs](https://docs.tempus.finance/docs/tempuspool/redemption).
```
Comment (RedFox20) [inline other/CodeArenaReadme.md:38]:
```suggestion
Users can redeem their funds into either Backing Tokens or Yield Bearing Tokens. You can find more in [redemption docs](https://docs.tempus.finance/docs/tempuspool/redemption).
```

File: other/CodeArenaReadme.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+All swaps between Principals and Yields are done via Tempus AMM. Information about our AMM can be found in [TempusAMM docs](https://docs.tempus.finance/docs/tempusamm) or in our [blog post](https://medium.com/tempusfinance/diving-into-tempus-amm-23a92cc6a2fc) about it.
+
+## Smart contracts
+All smart contracts are located in `contracts` directory. However, we added mocked implementations of underlying landing/staking protocols (Lido, Aave, and Compound). These mocks are not in focus for this contest, as they are only used for unit testing protocol related stuff. You can find these mocks in `contracts/mocks`.
```
Comment (RedFox20) [inline other/CodeArenaReadme.md:43]:
```suggestion
All smart contracts are located in `contracts/` directory. However, we added mocked implementations of underlying landing/staking protocols (Lido, Aave, and Compound). These mocks are not in focus for this contest, as they are only used for unit testing of the underlying protocols. You can find these mocks in `contracts/mocks/`.
```

File: other/CodeArenaReadme.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+| Glossary| |
+|-------------------------------|------------------------------------------------------|
+| Backing token| Any token that can be deposited to earn some interest via landing or staking. Currently we support ETH, DAI, and USDC |
+| YBT Yield bearing token| Landing or staking token, currently we support stETH, aTokens, and cTokens |
+| TempusPool| Holds all locked YieldBearingToken, and is used to mint Principals and Yields on deposits, and to burn them on redemption in exchange for YieldBearingToken |
+| Principals| Zero coupon bond redeemable for 1 Backing token after maturity |
+| Yields| Zero coupon bond redeemable after maturity for amount of Yield 1 Principal accrued for the lifetime of the pool |
+| Tempus AMM| Tempus AMM implementation based on Stable Pools from Balancer v2. It is used for trading Principals against Yields |
+| LP token| Token that represent share in Liquidity Pool |
```
[the remark this answers is not in the document]
Comment (mijovic) [reply other/CodeArenaReadme.md:20]: Would `| LP token| Token that represent share in Tempus AMM. Users get this token when they provide liquidity with Principals and Yields |` work?

File: other/CodeArenaReadme.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+Redemption has two different flows:
+- Early redemption (this is possible only with equal amount of Principals and Yields)
+- Mature redemption (user can deposit any combination of Principals and Yields after maturity)
+Redemption can be to backing or yield bearing token. You can find more in [redemption docs](https://docs.tempus.finance/docs/tempuspool/redemption).
+
+All swaps between Principals and Yields are done via Tempus AMM. Information about our AMM can be found in [TempusAMM docs](https://docs.tempus.finance/docs/tempusamm) or in our [blog post](https://medium.com/tempusfinance/diving-into-tempus-amm-23a92cc6a2fc) about it.
+
+## Smart contracts
+All smart contracts are located in `contracts` directory. However, we added mocked implementations of underlying landing/staking protocols (Lido, Aave, and Compound). These mocks are not in focus for this contest, as they are only used for unit testing protocol related stuff. You can find these mocks in `contracts/mocks`.
+Unit tests are located in `test` directory.
```
Comment (RedFox20) [inline other/CodeArenaReadme.md:44]:
```suggestion
Unit tests are located in `test` directory and integration tests are in `integration-tests` directory.
```

**Pull request closed without merging**